### PR TITLE
[Rendezvous] Add BLEEndPoint echo code to the esp32 demo app

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This patch add an echo mechanism for BLEEndPoint on the esp32 demo app.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by andy31415 — COMMENTED** [review]
How about making BLE a transport, in which case startServer could start a BLE processor.

File: examples/wifi-echo/server/esp32/main/Ble.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+using namespace ::chip;
+using namespace ::chip::Ble;
+
+void HandleBleMessageReceived(BLEEndPoint * endPoint, PacketBuffer * buffer)
+{
+    const size_t bufferLen = buffer->DataLength();
+    char msg[bufferLen];
```
Comment (kedars) [inline examples/wifi-echo/server/esp32/main/Ble.cpp:29]: Is there a max-limit on the Data Length here? The stack usage will vary depending upon this length received.

Comment (gerickson) [reply examples/wifi-echo/server/esp32/main/Ble.cpp:29]: +1, we should definitely avoid large stack-allocated buffers and objects wherever possible.

**Comment by woody-apple** [comment]
@vivien-apple can we sort out merge conflicts and feedback here?

**Comment by vivien-apple** [comment]
> How about making BLE a transport, in which case startServer could start a BLE processor.

I will do my best for the iOS side of things. My current plan is to build the BLEPlatformDelegate side of things under src/platform/Darwin and so the iOS side (#1460) will focus on trying to expose a higher level API to the app.

In the meantime I would like to keep this patch as if so I can start to build the platform BLE delegate and I will likely add more code to examples/chip-tool instead of the iOS app at first to validate that it works correctly. And I would appreciate to have this code on the esp32 to reply back to me. And fwiw this code is less invasive than the iOS side of thing.

**Review by BroderickCarlin — APPROVED** [review]

**Comment by vivien-apple** [comment]
> @vivien-apple can we sort out merge conflicts and feedback here?

Rebased. Still looking if I can find a #define somewhere for the limit

File: examples/wifi-echo/server/esp32/main/Ble.cpp (hunk 1 above)
Comment (rwalker-apple) [reply examples/wifi-echo/server/esp32/main/Ble.cpp:29]: the point of this buffer is logging?  how about only logging the first few bytes of the echo message instead?

Comment (rwalker-apple) [reply examples/wifi-echo/server/esp32/main/Ble.cpp:29]: we also want to sanitize the logging, or is that the job of the device layer?

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**